### PR TITLE
feat(sdk-coin-avaxp): implement generateKeyPair and signTransaction

### PR DESCRIPTION
**Pull request by jrafael-simtlix**
STLX-17279

### CODE REVIEW

**Comment by ducnguyen-bitgo** [comment]
@jrafael-simtlix Since there's already a `avaxp.ts` file [here](https://github.com/BitGo/BitGoJS/blob/STLX-17279-bitgo-sdk-implement-generate-key-pair-and-sign-transaction-in-bitgo-module/modules/bitgo/src/v2/coins/avaxp.ts). Could you git mv it to the sdk-coin-avaxp package so we could keep track of the git history? Also, once you move it to the coin package, could you import and re-export it [here](https://github.com/BitGo/BitGoJS/blob/STLX-17279-bitgo-sdk-implement-generate-key-pair-and-sign-transaction-in-bitgo-module/modules/bitgo/src/v2/coins/index.ts#L5) like how we did it for algo and cspr for backward compatibility? Thanks!

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 1]
```diff
@@ -0,0 +1,237 @@
+import * as AccountLib from '../../src/lib';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-avaxp/test/unit/avaxp.ts:1]: Let's change this to AvaxpLib for consistency

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+import { TestBitGo, TestBitGoAPI } from '@bitgo/sdk-test';
+import { Avaxp } from '../../src/';
+import { randomBytes } from 'crypto';
+import * as should from 'should';
+import { BitGoAPI } from '@bitgo/sdk-api';
+import { coins } from '@bitgo/statics';
+import * as testData from '../resources/avaxp';
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/test/unit/avaxp.ts:8]: Resources's merged in a preview PR.

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  });
+
+  // TODO(STLX-16570): explainTransaction
+  xdescribe('Explain Transaction', () => {
+    it('should explain a half signed AddValidator transaction', () => {
+      // TODO
+    });
+
+    it('should explain a signed AddValidator transaction', () => {
+      // TODO
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/test/unit/avaxp.ts]: Done

File: modules/sdk-coin-avaxp/test/unit/avaxp.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      }).getKeys();
+      const messageToSign = Buffer.from(randomBytes(32)).toString('hex');
+      await basecoin.signMessage(keyPair, messageToSign).should.be.rejectedWith('Invalid key pair options');
+    });
+  });
+
+  // TODO(STLX-16570): explainTransaction
+  xdescribe('Explain Transaction', () => {
+    it('should explain a half signed AddValidator transaction', () => {
+      // TODO
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/test/unit/avaxp.ts]: Done

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    try {
+      new Lib.KeyPair({ prv });
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  isWalletAddress(params: VerifyAddressOptions): boolean {
+    // TODO: Implement later.
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts]: Remove comment when avaxp from bitgo modules merge with this.

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  }
+  getFullName(): string {
+    return this._staticsCoin.fullName;
+  }
+  getBaseFactor(): string | number {
+    return Math.pow(10, this._staticsCoin.decimalPlaces);
+  }
+
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
+    // TODO: Implement when available on the SDK.
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts]: Remove comment when avaxp from bitgo modules merge with this.

File: modules/sdk-coin-avaxp/src/avaxp.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
+/**
+ * @prettier
+ */
+import * as Lib from './lib';
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/avaxp.ts]: Done, also in unittest

File: modules/sdk-coin-avaxp/src/index.ts [hunk 8]
```diff
@@ -1,26 +1,2 @@
...
-  coinName: string,
-  builderFactory: { new (_coinConfig: Readonly<CoinConfig>): T }
-): T {
-  const coinConfig = coins.get(coinName);
-  const factory = new builderFactory(coinConfig);
-  // coinBuilderMap[coinName] = factory;
-  // coinBuilderMap[coinName] = builderFactory; // For now register the constructor function until reimplement getBuilder method
-  return factory;
-}
+export * from './avaxp';
```
[the remark this answers is not in the document]
Comment (jrafael-simtlix) [reply modules/sdk-coin-avaxp/src/index.ts:1]: Add it

File: modules/bitgo/package.json [hunk 9]
```diff
@@ -51,6 +51,7 @@
     "@bitgo/blockapis": "^1.0.1",
     "@bitgo/sdk-api": "^1.0.1",
     "@bitgo/sdk-coin-algo": "^1.0.0",
+    "@bitgo/sdk-coin-avaxp": "^1.0.0",
```
Comment (mmcshinsky-bitgo) [inline modules/bitgo/package.json:54]: Add coin to tsconfig as well

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/bitgo/package.json (hunk 9 above)
Comment (shiksb) [reply modules/bitgo/package.json:54]: added

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**